sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/Credential/ServicePrincipalCredentialEntity.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+
+namespace Azure.AI.MetricsAdvisor.Models
+{
+    /// <summary>
+    /// Authenticates to an Azure service via service principal.
+    /// </summary>
+    [CodeGenModel("ServicePrincipalCredential")]
+    [CodeGenSuppress(nameof(ServicePrincipalCredentialEntity), typeof(string), typeof(ServicePrincipalParam))]
+    public partial class ServicePrincipalCredentialEntity
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ServicePrincipalCredentialEntity"/> class.
+        /// </summary>
+        /// <param name="name">A custom unique name for this <see cref="ServicePrincipalCredentialEntity"/> to be displayed on the web portal.</param>
+        /// <param name="clientId">The client ID of the service principal used for authentication.</param>
+        /// <param name="clientSecret">The client Secret of the service principal used for authentication.</param>
+        /// <param name="tenantId">The tenant ID of the service principal used for authentication.</param>
+        public ServicePrincipalCredentialEntity(string name, string clientId, string clientSecret, string tenantId) : base(name)
+        {
+            Argument.AssertNotNullOrEmpty(clientId, nameof(clientId));
+            Argument.AssertNotNullOrEmpty(clientSecret, nameof(clientSecret));
+            Argument.AssertNotNullOrEmpty(tenantId, nameof(tenantId));
+
+            DataSourceCredentialType = DataSourceCredentialType.ServicePrincipal;
+            ClientId = clientId;
+            ClientSecret = clientSecret;
+            TenantId = tenantId;
+        }
+
+        internal ServicePrincipalCredentialEntity(DataSourceCredentialType dataSourceCredentialType, string id, string name, string description, ServicePrincipalParam parameters) : base(dataSourceCredentialType, id, name, description)
+        {
+            DataSourceCredentialType = dataSourceCredentialType;
+            ClientId = parameters.ClientId;
+            ClientSecret = parameters.ClientSecret;
+            TenantId = parameters.TenantId;
+        }
+
+        /// <summary>
+        /// The client ID of the service principal used for authentication.
+        /// </summary>
+        public string ClientId { get; set; }
+
+        /// <summary>
+        /// The client Secret of the service principal used for authentication.
+        /// </summary>
+        public string ClientSecret { get; set; }
```

Comment:
is it safe for us to get this value back and leave it as a string?
can anyone call this API? or only people with admin or elevated permission?

Reply:
This is a great question and has generated some discussion in the last couple of days. The service decided to not return secret values even during a Get operation. This change is not deployed yet, though. I want to do something similar to what we do in our key credential types (making it volatile, hiding it), but it will affect some other types we have as well.

Reply:
so what your saying is that it will happen in a future PR?

Reply:
Correct. The service changed a few things in the latest deployment (and broke 172 tests in total), and I wanted to take the new behavior in consideration before choosing the design. I'll update the credential handling with the design changes.

Reply:
Updated PR to hide credentials.